packages/@okta/vuepress-site/docs/reference/api/roles/index.md

```diff
@@ -1277,6 +1277,808 @@ curl -v -X DELETE \
+            "hints": {
+                "allow": [
+                    "POST"
+                ]
+            }
+        },
+        "self": {
+            "href": "http://rain.okta1.com:1802/api/v1/roles/SUPER_ADMIN/subscriptions/IWA_AGENT"
+        }
+    }
+}
+```
+
+#### Subscribe to a specific type
+
+<ApiLifecycle access="ea" />
+
+<ApiOperation method="post" url="/api/v1/roles/${roleType}/subscriptions/${notificationType}/subscribe" />
+
+Make a Role subscribes to a specific type
+
+##### Request parameters
+
+
+| Parameter        | Description                                            | Param Type  | DataType                                 | Required  |
+| :--------------- | :----------------------------------------------------- | :---------- | :--------------------------------------- | :-------- |
+| roleType         | `type` of a Role                                       | URL         | [Role Type](#role-types)                 | TRUE      |
+| notificationType | `type` of a notification                               | URL         | [Notification Type](#notification-type) | TRUE      |
+
+##### Response parameters
+
+
+``` http
+HTTP/1.1 200 OK
+```
+
+##### Request example
+
+
+```bash
+curl -v -X POST \
+-H "Accept: application/json" \
+-H "Content-Type: application/json" \
+-H "Authorization: SSWS ${api_token}" \
+"https://${yourOktaDomain}/api/v1/roles/SUPER_ADMIN/subscriptions/IWA_AGENT/subscribe"
+```
+
+##### Response example
+
+
+``` http
+HTTP/1.1 200 OK
+```
+
+#### Unsubscribe to a specific type
+
+<ApiLifecycle access="ea" />
+
+<ApiOperation method="post" url="/api/v1/roles/${roleType}/subscriptions/${notificationType}/unsubscribe" />
+
+Make a Role unsubscribes to a specific type
+
+##### Request parameters
+
+
+| Parameter        | Description                                            | Param Type  | DataType                                 | Required  |
+| :--------------- | :----------------------------------------------------- | :---------- | :--------------------------------------- | :-------- |
+| roleType         | `type` of a Role                                       | URL         | [Role Type](#role-types)                 | TRUE      |
+| notificationType | `type` of a notification                               | URL         | [Notification Type](#notification-type)  | TRUE      |
+
+
+##### Response parameters
+
+
+``` http
+HTTP/1.1 200 OK
+```
+
+##### Request example
+
+
+```bash
+curl -v -X POST \
+-H "Accept: application/json" \
+-H "Content-Type: application/json" \
+-H "Authorization: SSWS ${api_token}" \
+"https://${yourOktaDomain}/api/v1/roles/SUPER_ADMIN/subscriptions/IWA_AGENT/unsubscribe"
+```
+
+##### Response example
+
+
+``` http
+HTTP/1.1 200 OK
+```
+
+#### List subscriptions of a User
+
+<ApiLifecycle access="ea" />
+
+<ApiOperation method="get" url="/api/v1/users/${userId}/subscriptions" />
+
+List subscriptions of a User. We only support operations for current user right now.
```

Comment:
what happens if this is called with a non-admin user id?

Reply:
If will first check the permission then valid the user id is the current user or not. So if the user doesn't have the permission, the call would fail for access denial